### DISCUSSION TITLE
How to change inputs to task inside a Do While Task

### DISCUSSION THREAD

**Discussion by Priyank0711** [Netflix/conductor#2696] [Q&A]
I am trying to execute a task inside a Do While Task. 
Here I have to update the inputs to the task with every iteration.
How can we do that?

**Comment by apanicker-nflx** [comment]
@manan164 @rickfish Can you help with this?

**Reply by manan164** [reply]
Hi @Priyank0711 , Are you trying to say,
1. Update input every iteration dynamically - This is currently not supported.
2. Update input based on the last iteration output - This is supported. Please share the workflow definition. I can help with this.

**Comment by amoroziuk** [comment]
Hi @apanicker-nflx , @manan164, I have a bit similar issue: I can't modify input variables dynamically or use workflow's variables inside of DO_WHILE condition.
Here is the description of my case:
I need to build a workflow like this:
<img width="255" alt="image" src="https://github.com/Netflix/conductor/assets/49148761/55f5b2d5-2f76-481b-9b8e-e9e800e061c9">
So, this is a workflow with 2 approval steps (`'Task2'` and `'Task3'`), in my case approval step contains two separate tasks: SIMPLE (`'Task2'`, `'Task3'`) and SWITCH (`'Task2 approved'`, `'Task3 approved'`). This workflow should support the retry policy and be able to redirect workflow execution to the previous step (if `'Task2'` was rejected -> go to `'Task1'` again; if `'Task3'` was rejected - > go to `'Task2'`). I was trying to achieve this by using multiple DO_WHILE tasks, here is the simplified workflow for this case:
```json
{
  "createTime": 1692365021311,
  "updateTime": 1692368195606,
  "accessPolicy": {},
  "name": "SimpleLoop",
  "description": "Edit or extend this sample workflow. Set the workflow name to get started",
  "version": 8,
  "tasks": [
    {
      "name": "Loop Task",
      "taskReferenceName": "LoopTask1",
      "inputParameters": {
        "value": "${workflow.input.value}"
      },
      "type": "DO_WHILE",
      "startDelay": 0,
      "optional": false,
      "asyncComplete": false,
      "loopCondition": "if ( $.second['approved']) { false; } else { true; }",
      "loopOver": [
        {
          "name": "first",
          "taskReferenceName": "first",
          "description": "First Task",
          "inputParameters": {},
          "type": "SIMPLE",
          "startDelay": 0,
          "optional": false,
          "asyncComplete": false,
          "retryCount": 0,
          "evaluatorType": "javascript"
        },
        {
          "name": "Loop Task2",
          "taskReferenceName": "LoopTask2",
          "inputParameters": {
            "value": "${workflow.input.value}",
            "outerItaration": "$.LoopTask1['iteration']"
          },
          "type": "DO_WHILE",
          "startDelay": 0,
          "optional": false,
          "asyncComplete": false,
          "loopCondition": "if ($.second['approved']) { if (!$.third['approved']) { true; } else { false; } } else { false; }",
          "loopOver": [
            {
              "name": "second",
              "taskReferenceName": "second",
              "description": "second Task",
              "inputParameters": {},
              "type": "SIMPLE",
              "startDelay": 0,
              "optional": false,
              "asyncComplete": false,
              "retryCount": 0,
              "evaluatorType": "javascript"
            },
            {
              "name": "third",
              "taskReferenceName": "third",
              "description": "third Task",
              "inputParameters": {},
              "type": "SIMPLE",
              "startDelay": 0,
              "optional": false,
              "asyncComplete": false,
              "retryCount": 0,
              "evaluatorType": "javascript"
            }
          ]
        }
      ]
    }
  ],
  "inputParameters": [],
  "outputParameters": {
  },
  "schemaVersion": 2,
  "restartable": true,
  "workflowStatusListenerEnabled": false,
  "ownerEmail": "example@email.com",
  "timeoutPolicy": "ALERT_ONLY",
  "timeoutSeconds": 0,
  "variables": {},
  "inputTemplate": {}
}
```

I'm using output parameters to mark tasks approved or rejected (in this case just using 'approved' output variable which is boolean).
The test script here is:
1. Start workflow -> `'first'` task should be scheduled
2. Complete `'first'` task with `'approved: true'` as an output (approve the task): 
`{
  "workflowInstanceId": "{your_instance_id}",
  "taskId": "{your_task_id}",
  "status": "COMPLETED",
  "outputData": {
    "approved": true
  }
}`
3. `'second'` task should be scheduled 
4. Complete `'second'` task with `'approved: false'` (reject the task) -> `'third'` task should be scheduled
5. Reject `'third'` task -> `'first'` task should be scheduled again
6. Approve `'first'` task again

In this case nothing will happen after 6-th step, but it should schedule `'second'` task again.
Could you please add your recommendation on how can I implement this workflow (maybe there is another way to do that)? Maybe there is any other way how to dynamically check the condition for loops?

**Reply by amoroziuk** [reply]
@manan164 Is there any other way to go to a specific task instead of using loops?
I saw this conversation: https://github.com/Netflix/conductor/issues/1031 is there any news related to adding this task type?
